drivers/display/ssd16xx.c

```diff
@@ -106,12 +108,13 @@ static inline int ssd16xx_write_cmd(struct ssd16xx_data *driver,
 
 static inline void ssd16xx_busy_wait(struct ssd16xx_data *driver)
 {
-	u32_t val = 0U;
+	int pin = gpio_pin_get(driver->busy, SSD16XX_BUSY_PIN);
 
-	gpio_pin_read(driver->busy, SSD16XX_BUSY_PIN, &val);
-	while (val) {
+	__ASSERT(pin >= 0, "Failed to get pin level");
+	while (pin) {
+		__ASSERT(pin >= 0, "Failed to get pin level");
```

Comment:
We could remove the first assert, the second one will catch the error, if there is one.